### PR TITLE
Translated Test-Rezepte

### PR DESCRIPTION
**Pull request by michael-schantin**

### CODE REVIEW

File: content/docs/testing-recipes.md [hunk 1]
```diff
@@ -415,17 +415,17 @@ it("changes value when clicked", () => {
 });
 ```
 
-Different DOM events and their properties are described in [MDN](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent). Note that you need to pass `{ bubbles: true }` in each event you create for it to reach the React listener because React automatically delegates events to the root.
+Verschiedene DOM-Ereignisse und ihre Eigenschaften werden in [MDN](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent) beschrieben. Beachten Sie, dass Sie `{ bubbles: true }` in jedem von Ihnen erstellten Ereignis übergeben müssen, damit es den React-Listener erreicht, da React Ereignisse automatisch an den Stamm delegiert.
```
Comment (ph1p) [inline content/docs/testing-recipes.md:418]:
```suggestion
Verschiedene DOM-Events und ihre Eigenschaften werden in [MDN](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent) beschrieben. Beachte, dass Du `{ bubbles: true }` in jedem von deinen erstellten Events übergeben musst, damit es den React-Listener erreicht, da React-Events automatisch an den Stamm (engl. root) delegiert werden.
```

File: content/docs/testing-recipes.md [hunk 2]
```diff
@@ -415,17 +415,17 @@ it("changes value when clicked", () => {
...
-> React Testing Library offers a [more concise helper](https://testing-library.com/docs/dom-testing-library/api-events) for firing events.
+> Die React Testing Library bietet einen [übersichtlicheren Helfer](https://testing-library.com/docs/dom-testing-library/api-events) für das Auslösen von Ereignissen.
 
 ---
 
-### Timers {#timers}
+### Timer {#timers}
 
-Your code might use timer-based functions like `setTimeout` to schedule more work in the future. In this example, a multiple choice panel waits for a selection and advances, timing out if a selection isn't made in 5 seconds:
+Ihr Code könnte zeitgeberbasierte Funktionen wie `setTimeout` verwenden, um mehr Arbeit in der Zukunft zu planen. In diesem Beispiel wartet ein Multiple-Choice-Feld auf eine Auswahl und rückt vor, wobei eine Zeitüberschreitung eintritt, wenn innerhalb von 5 Sekunden keine Auswahl getroffen wird:
```
Comment (ph1p) [inline content/docs/testing-recipes.md:428]:
```suggestion
Dein Code könnte zeitgeberbasierte Funktionen wie `setTimeout` verwenden, um mehr Arbeit in der Zukunft zu planen. In diesem Beispiel wartet ein Multiple-Choice-Feld auf eine Auswahl und rückt vor, wobei eine Zeitüberschreitung eintritt, wenn innerhalb von 5 Sekunden keine Auswahl getroffen wird:
```

File: content/docs/testing-recipes.md [hunk 3]
```diff
@@ -538,15 +538,14 @@ it("should accept selections", () => {
 });
 ```
 
-You can use fake timers only in some tests. Above, we enabled them by calling `jest.useFakeTimers()`. The main advantage they provide is that your test doesn't actually have to wait five seconds to execute, and you also didn't need to make the component code more convoluted just for testing.
+Sie können gefälschte Timer nur in einigen Tests verwenden. Oben haben wir sie aktiviert, indem wir `jest.useFakeTimers()` aufgerufen haben. Der Hauptvorteil, den sie bieten, besteht darin, dass Ihr Test nicht wirklich fünf Sekunden auf die Ausführung warten muss, und Sie mussten den Komponentencode auch nicht nur zum Testen komplizierter machen.
```
Comment (ph1p) [inline content/docs/testing-recipes.md:541]:
```suggestion
Du kannst Fake-Timer nur in einigen Tests verwenden. Oben haben wir sie aktiviert, indem wir `jest.useFakeTimers()` aufgerufen haben. Der Hauptvorteil, den sie bieten, besteht darin, dass dein Test nicht wirklich fünf Sekunden auf die Ausführung warten  und du den Komponentencode auch nicht nur zum Testen komplizierter machen musst.
```

File: content/docs/testing-recipes.md [hunk 4]
```diff
@@ -538,15 +538,14 @@ it("should accept selections", () => {
...
 
 ---
 
-### Snapshot Testing {#snapshot-testing}
-
-Frameworks like Jest also let you save "snapshots" of data with [`toMatchSnapshot` / `toMatchInlineSnapshot`](https://jestjs.io/docs/en/snapshot-testing). With these, we can "save" the rendered component output and ensure that a change to it has to be explicitly committed as a change to the snapshot.
+### Snapshot-Testen {#snapshot-testing}
 
-In this example, we render a component and format the rendered HTML with the [`pretty`](https://www.npmjs.com/package/pretty) package, before saving it as an inline snapshot:
+Mit Frameworks wie Jest können Sie mit [`toMatchSnapshot` / `toMatchInlineSnapshot`](https://jestjs.io/docs/en/snapshot-testing) auch "Schnappschüsse" von Daten speichern. Mit diesen können wir die gerenderte Komponentenausgabe "speichern" und sicherstellen, dass eine Änderung daran explizit als Änderung am Snapshot festgeschrieben werden muss.
```
Comment (ph1p) [inline content/docs/testing-recipes.md:547]:
```suggestion
Mit Frameworks wie Jest kannst Du mit [`toMatchSnapshot` / `toMatchInlineSnapshot`](https://jestjs.io/docs/en/snapshot-testing) auch "Schnappschüsse" von Daten speichern. Mit diesen können wir die gerenderte Komponentenausgabe "speichern" und sicherstellen, dass eine Änderung daran explizit als Änderung am Snapshot festgeschrieben werden muss.
```

File: content/docs/testing-recipes.md [hunk 5]
```diff
@@ -599,13 +598,13 @@ it("should render a greeting", () => {
...
 
 ---
 
-### Multiple Renderers {#multiple-renderers}
+### Mehrere Renderer {#multiple-renderers}
 
-In rare cases, you may be running a test on a component that uses multiple renderers. For example, you may be running snapshot tests on a component with `react-test-renderer`, that internally uses `render` from `react-dom` inside a child component to render some content. In this scenario, you can wrap updates with `act()`s corresponding to their renderers.
+In seltenen Fällen führen Sie möglicherweise einen Test für eine Komponente aus, die mehrere Renderer verwendet. Beispielsweise können Sie Snapshot-Tests auf einer Komponente mit `react-test-renderer` ausführen, die intern `render` von `react-dom` innerhalb einer untergeordneten Komponente verwendet, um einige Inhalte zu rendern. In diesem Szenario können Sie Aktualisierungen mit `act()`s umschließen, die ihren Renderern entsprechen.
```
Comment (ph1p) [inline content/docs/testing-recipes.md:607]:
```suggestion
In seltenen Fällen führen Sie möglicherweise einen Test für eine Komponente aus, die mehrere Renderer verwendet. Beispielsweise kannst du Snapshot-Tests auf einer Komponente mit `react-test-renderer` ausführen, die intern `render` von `react-dom` innerhalb einer untergeordneten Komponente verwendet, um einige Inhalte zu rendern. In diesem Szenario können Sie Aktualisierungen mit `act()`s umschließen, die ihren Renderern entsprechen.
```

File: content/docs/testing-recipes.md [hunk 6]
```diff
@@ -622,6 +621,6 @@ expect(root).toMatchSnapshot();
 
 ---
 
-### Something Missing? {#something-missing}
+### Etwas fehlt? {#something-missing}
 
-If some common scenario is not covered, please let us know on the [issue tracker](https://github.com/reactjs/reactjs.org/issues) for the documentation website.
+Wenn ein gängiges Szenario nicht abgedeckt ist, teilen Sie uns dies bitte auf der [Problemverfolgungswebsite](https://github.com/reactjs/reactjs.org/issues) für die Dokumentation mit.
```
Comment (ph1p) [inline content/docs/testing-recipes.md:626]:
```suggestion
Wenn ein gängiges Szenario nicht abgedeckt ist, teile uns dies bitte im [Issue-Tracker](https://github.com/reactjs/reactjs.org/issues) für die Dokumentation mit.
```

File: content/docs/testing-recipes.md [hunk 7]
```diff
@@ -107,41 +107,41 @@ import Hello from "./hello";
...
   unmountComponentAtNode(container);
   container.remove();
   container = null;
 });
 
-it("renders with or without a name", () => {
+it("Rendert mit oder ohne Namen", () => {
   act(() => {
-    render(<Hello />, container);
+    render(<Hallo />, container);
```
Comment (ph1p) [inline content/docs/testing-recipes.md:124]:
```suggestion
    render(<Hello />, container);
```
We decided to translate only comments and plain text to keep it consistent. Not the code itself.

**Review by ph1p — CHANGES_REQUESTED** [review]
Thank you! :) I have added a few suggestions

**Comment by michael-schantin** [comment]
I agree to all changes, is there anything else I need to do to make the PR gemer

**Comment by ph1p** [comment]
Nope :) Thank you very much!

**Review by ph1p — APPROVED** [review]

**Pull request merged**